aip/0231.md

```diff
@@ -0,0 +1,86 @@
+---
+aip:
+  id: 231
+  state: reviewing
+  created: 2019-06-18
+permalink: /231
+redirect_from:
+  - /0231
+---
+
+# Batch Get
+
+Some APIs need to allow users to get a specific set of resources at a consistent
+time point (e.g. using a read transaction). A batch get method provides this
+functionality.
+
+## Guidance
+
+APIs **may** support Batch Get using the following pattern:
+
+```proto
+rpc BatchGetBooks(BatchGetBooksRequest) returns (BatchGetBooksResponse) {
+  option (google.api.http) = {
+    get: "/v1/shelves/-/books:batchGet"
+  };
+}
+```
+
+- The RPC's name **must** begin with `BatchGet`. The remainder of the
+  RPC name **should** be the plural form of the resource being retrieved.
+- The request and response messages **must** match the RPC name, with `-Request`
+  and `-Response` suffixes.
+- The HTTP verb **must** be `GET`.
```

Comment:
**must** be `GET` with the exception for URL length restrictions. 

Or something. To say **must** be `Get`, followed 3 points later saying use `Post` is jarring.

Reply:
It's not really contradictory. The verb must be specified as GET in the annotation, but users *can* make a POST request instead. That said, I'm not sure it's particularly worth including the part about users making a POST request, as that's not advice for the API producer as such.